app/lib/common/providers/notifiers/chat_notifiers.dart

```diff
@@ -86,54 +86,66 @@ class ChatRoomsListNotifier extends StateNotifier<List<Convo>> {
 
   List<Convo> listCopy() => List.from(state, growable: true);
 
+  void _sortList(List<Convo> list) {
+    list.sort((a, b) => b.latestMessageTs().compareTo(b.latestMessageTs()));
+  }
+
   void _handleDiff(ConvoDiff diff) {
     switch (diff.action()) {
       case 'Append':
         final newList = listCopy();
         List<Convo> items = diff.values()!.toList();
         newList.addAll(items);
+        _sortList(newList);
         state = newList;
         break;
       case 'Insert':
         Convo m = diff.value()!;
         final index = diff.index()!;
         final newList = listCopy();
         newList.insert(index, m);
+        _sortList(newList);
         state = newList;
         break;
       case 'Set':
         Convo m = diff.value()!;
         final index = diff.index()!;
         final newList = listCopy();
         newList[index] = m;
+        _sortList(newList);
         state = newList;
         break;
       case 'Remove':
         final index = diff.index()!;
         final newList = listCopy();
         newList.removeAt(index);
+        _sortList(newList);
         state = newList;
         break;
```

Comment:
I like the idea, but this _won't work_. The `ConvoDiff` is a sorted list using indixes, you are changing the list that the next diff is going to be applied on. This is particuarly problematic for  cases like this `remove` here, where an item is supposed to be removed based on the index. Because you resorted the list before, we will be removing the wrong item.